telethon/events/callbackquery.py

```diff
@@ -78,6 +98,12 @@ def filter(self, event):
             elif event.query.data != self.data:
                 return None
 
+        if self.pattern:
+            match = self.pattern(event.message.message or '')
```

Comment:
Do we want to match on the message, or should pattern be a regex for the data?

Reply:
The pattern not the message.